lib/Runtime/Types/Type.h

```diff
@@ -62,6 +63,14 @@ namespace Js
         inline BOOL SkipsPrototype() const { return (this->flags & TypeFlagMask_SkipsPrototype) != 0 ; }
         inline BOOL CanHaveInterceptors() const { return (this->flags & TypeFlagMask_CanHaveInterceptors) != 0; }
         inline BOOL IsFalsy() const { return flags & TypeFlagMask_IsFalsy; }
+        inline BOOL HasBeenCached() const { return flags & TypeFlagMask_HasBeenCached; }
+        inline void SetHasBeenCached(const bool hasBeenCached)
+        {
+            // Once set, this flag should never be reset.
```

Comment:
When we delete a property, we change the type of the object to invalidate any caches which may have cached its type. We don't have to change the type if it wasn't cached anywhere.
But won't it mean deleteProperty on invalidated type change the type even though it shouldn't because the type is no longer cached?

---
In reply to: [78816821](https://github.com/Microsoft/ChakraCore/pull/1542#discussion_r78816821) [](ancestors = 78816821,78790053)